### DISCUSSION TITLE
Air quality sensor reading period

### DISCUSSION THREAD

**Discussion by Evaldas1979** [ESPresense/ESPresense#787] [Q&A]
I have connected the air quality sensor SGP30 to I2C bus. ESPresense device transmits CO2 and TVOC values to MQTT server every second.
My question is how to change this period to much longer?

**Comment by Terastar-Paperclip** [comment]
Flagging this for the core team to confirm whether the SGP30 publish interval is configurable today or would need a firmware change — don't want to guess and send you down the wrong path. Will follow up here.

**Comment by Terastar-Paperclip** [comment]
Tech Lead here, checked current source. Good news: this is already fixed on current firmware, just not the way you'd expect. `SensirionSGP30.cpp` samples the sensor every 1s internally (required by the SGP30's baseline-compensation algorithm — it needs regular reads to stay calibrated) but only **publishes** to MQTT once every 60s (`reportInterval = 60000`, same pattern used for the SCD4x). So you're not actually getting 1 Hz MQTT traffic on current firmware.

That interval isn't exposed as a user setting today though — it's a hardcoded constant, unlike the I2C bus/address for this sensor which *are* `HeadlessWiFiSettings` entries. If you want something other than 60s, that needs a firmware change (small one — same pattern as `SGP30_I2c` already in that file). Marking **good first issue** for whoever wants to pick it up; happy to review.
